R/init.R

```diff
@@ -237,25 +206,94 @@ init <- function(data,
     id <- character(0)
   }
   ns <- NS(id)
+
   # Note: UI must be a function to support bookmarking.
-  res <- list(
-    ui = function(request) {
-      ui_teal(
-        id = ns("teal"),
-        modules = modules,
-        title = title,
-        header = header,
-        footer = footer
-      )
-    },
-    server = function(input, output, session) {
-      if (!is.null(landing_popup)) {
-        do.call(landing_popup$server, c(list(id = "landing_module_shiny_id"), landing_popup$server_args))
+  res <- structure(
+    list(
+      ui = function(request) {
+        fluidPage(
+          title = tags$div(
+            id = "teal-app-title",
+            tags$head(
+              tags$title("teal app"),
+              tags$link(
+                rel = "icon",
+                href = "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/nest.png",
+                sizes = "any"
+              )
+            )
+          ),
+          tags$header(
+            id = "teal-header",
+            tags$div(id = "teal-header-content")
+          ),
+          ui_teal(
+            id = "teal",
+            modules = modules
+          ),
+          tags$footer(
+            id = "teal-footer",
+            tags$div(id = "teal-footer-content"),
+            ui_session_info("teal-footer-session_info")
+          )
+        )
+      },
+      server = function(input, output, session) {
+        srv_teal(id = "teal", data = data, modules = modules, filter = deep_copy_filter(filter))
+        srv_session_info("teal-footer-session_info")
       }
-      srv_teal(id = ns("teal"), data = data, modules = modules, filter = deep_copy_filter(filter))
-    }
+    ),
+    class = "teal_app"
   )
 
+  if (lifecycle::is_present(title)) {
+    checkmate::assert_multi_class(title, c("shiny.tag", "shiny.tag.list", "html", "character"))
+    lifecycle::deprecate_warn(
```

Comment:
I thin it's better to throw deprecation warning first before making assertions.

Suppose I provide the `title` argument and I failed assertions. I am putting a lot of effort into "fixing it" just to see that this is deprecated.

Reply:
Agreed